test/functional/Java21Only/src/org/openj9/test/jep442/downcall/PrimitiveTypeTests1.java

```diff
@@ -44,6 +44,8 @@
  */
 @Test(groups = { "level.sanity" })
 public class PrimitiveTypeTests1 {
+	private static final boolean isZos = "z/OS".equalsIgnoreCase(System.getProperty("os.name"));
+	private static final String charset = isZos ? "IBM1047" : "UTF-8";
```

Comment:
I think the test should continue to exercise `allocateUtf8String()` on other platforms; a helper method will make that easier:
```java
	static MemorySegment allocateString(String string) {
		MemorySegment segment;

		if (isZos) {
			byte[] bytes = string.getBytes("IBM1047");
			byte[] withNul = Arrays.copyOf(bytes, bytes.length + 1);

			segment = arena.allocateArray(JAVA_BYTE, withNul);
		} else {
			segment = arena.allocateUtf8String(str);
		}

		return segment;
	}
```
So the only other changes needed are to replace uses of `arena.allocateUtf8String()` with `allocateString()`.
The method can also be used in `PrimitiveTypeTests2.java`.

Reply:
Updated the code against the suggestion above and verified on z/OS  & zLinux.